emds: emds doc and emds usage in light_ctrl sample improvement (#23969)
nrfconnect/sdk-nrf: samples/bluetooth/mesh/light_ctrl/README.rst

To build this sample with support for emergency data storage (EMDS), set :makevar:`EXTRA_CONF_FILE` to :file:`overlay-emds.conf` using the respective :ref:`CMake option <cmake_options>`.
This will save replay protection list (RPL) data and some of the :ref:`bt_mesh_lightness_srv_readme` data to the emergency data storage instead of to the :ref:`settings_api`.

> The EMDS storing data API is called from the interrupt service routine (:ref:`ISR <zephyr:interrupts_v2>`) with priority 0 to prevent storing procedure pre-emption.

Reviewer: @peknis will probably anyway correct this, I'll try to suggest something to clarify this:

```suggestion
The EMDS storing data API is called from the interrupt service routine (:ref:`ISR <zephyr:interrupts_v2>`) with priority 0 to prevent pre-emption of the storing procedure.
```

> As a part of the interrupt routine, the Multiprotocol Service Layer (:ref:`MPSL <mpsl_lib>`) is uninitialized to ensure that no radio activity is ongoing during the EMDS storing procedure (only if radio exists).

Reviewer: Not clear what `only if radio exists` means here. If it is an implicit reference to emds.rst doc, then needs to be removed, because in this sample radio always exists. Otherwise, needs to be clarified.

> As a part of the interrupt routine, the Multiprotocol Service Layer (:ref:`MPSL <mpsl_lib>`) is uninitialized to ensure that no radio activity is ongoing during the EMDS storing procedure (only if radio exists as a peripheral module for core where EMDS is run).

Reviewer 2: ```suggestion
As a part of the interrupt routine, the Multiprotocol Service Layer (:ref:`MPSL <mpsl_lib>`) is uninitialized to ensure that no radio activity is ongoing during the EMDS storing procedure (only if radio is present as a peripheral module in the core running the EMDS).
```

> After the EMDS storing procedure is complete, sample leaves the MPSL uninitialized until the next system reset.

Reviewer 2: ```suggestion
After the EMDS storing procedure has completed, the sample leaves the MPSL uninitialized until the next system reset.
```

> After the EMDS storing procedure has completed, the sample leaves the MPSL uninitialized until the next system reset.

Reviewer: ```suggestion
After the storing procedure completes, the sample leaves the MPSL uninitialized until the next system reset.
```

> Since EMDS callback about data storing finished is called from the same interrupt service routine, it is not possible to use the MPSL API and API that uses the MPSL in the EMDS callback.

Reviewer 2: ```suggestion
Because the EMDS callback about data storing completed is called from the same interrupt service routine, it is not possible to use the MPSL API and API that uses the MPSL in the EMDS callback.
```

> Because the EMDS callback about data storing completed is called from the same interrupt service routine, it is not possible to use the MPSL API and API that uses the MPSL in the EMDS callback.

Reviewer: ```suggestion
Because the EMDS "data storing completed" callback is invoked from the same interrupt service routine, it is not possible to use the MPSL API or any API that depends on MPSL within this callback.
```
